### ISSUE TITLE
[Bug Report] Magic.from_file() chokes on pipe file

### ISSUE THREAD

**Issue by Shad0** [ahupp/python-magic#261]
The Magic class's from_file() method begins with an elegant way of catching certain errors. Lines 110-113 in \_\_init\_\_.py:

```
    def from_file(self, filename):
        # raise FileNotFoundException or IOError if the file does not exist
        with _real_open(filename):
            pass
```

_real_open() is just Python's builtin [open()](https://docs.python.org/3/library/functions.html#open) function. Lines 28-29 in \_\_init\_\_.py:

```
# avoid shadowing the real open with the version from compat.py
_real_open = open
```

This causes a problem in an admittedly obscure situation: where the target file is a pipe file (mime type "inode/fifo"). The open() function opens the pipe for reading, and then sits there forever waiting for input that never comes. Which means from_file() also just sits there forever, never calling libmagic.magic_file() or returning any value or throwing any error.

(This is my first time reporting an issue on GitHub. Apologies if I've failed to follow the proper procedure!)

**Comment by ahupp** [comment]
Very sorry for the extremely slow response.   This was fixed as a happy side effect of 722995466fda1451570c555e688819b75f1b8e65.  Your issue report was fine :)

**Comment by thiswillbeyourgithub** [comment]
Hi. I believe I have the same bug when developping [wdoc](https://github.com/thiswillbeyourgithub/wdoc/). I can confirm the bug is fixed in 0.4.28 but the latest pypi release is 0.4.28. 

Is it possible to do a new release by any change? It's somewhat harder to specify in my setup.py to install python-magic via git instead of pypi.
